[SherniceSng] iP

SherniceSng opened the pull request:
### **ChaCha the Chatbot**

> "Action is the foundational key to all success." -- Pablo Picasso ([source](https://dansilvestre.com/productivity-quotes/))

ChaCha is your one-stop solution to help you get organised. Its functions include:

- _text_-based
- super _easy_ to learn
- _simple_ to use
- very _efficient_

To set up, all you need to do is:

1. fork it from [here](https://github.com/SherniceSng/ip)
2. clone and run it
3. add whatever tasks you want
4. let ChaCha do the rest of the work for you! 🫡

**Features**:

- [ ] ~~Forget~~ Manage tasks, whether it is todo, deadlines or events

Practice Java too by inputting the `main` method below:

```java
public class Main {
   public static void main(String[] args) {
      print("Hello World");
   }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SherniceSng committed:
Renamed bot. Added greet and exit.

SherniceSng committed:
Added statements to allow echoing of commands and exit with command bye.

SherniceSng committed:
Added ListOfTask. Added add and list functionality.

SherniceSng committed:
Added Task class. Added mark as done / undone functionality.

SherniceSng committed:
Added ToDoTask, DeadlineTask, EventTask.

SherniceSng committed:
Change the way to do tasks are added, allowing tasks that consist of several words to be inputted.

SherniceSng committed:
Add ChaChaException.java, handle various exceptions.

SherniceSng committed:
Update exception handling by including more exception cases considered, and add delete functionality.

SherniceSng committed:
Update EXPECTED.TXT and input.txt with the test cases considered, and have minor changes to the formatting of ChaCha.java and ListOfTask.java.

SherniceSng committed:
Update ChaCha.java to check if data file chacha.txt exists and create one otherwise

SherniceSng committed:
ChaCha.java: Update to fix reading of chacha.txt file
ListOfTask.java: Update with copyFile method which will read and save each task into the ArrayList tasks
Task.java: Update constructor to take in isDone as argument
DeadlineTask.java: Update constructor to take in isDone as argument
ToDoTask.java: Update constructor to take in isDone as argument
EventTask.java: Update constructor to take in isDone and date as argument

SherniceSng committed:
ListOfTask.java: Fix reading file bug and add update chacha.txt functionlity after respective command
DeadlineTask.java: Change access modifiers for fields
EventTask.java: Change access modifiers for fields
Task.java: Add fields needed to update chacha.txt file

SherniceSng committed:
Merge branch 'branch-Level-7'

SherniceSng committed:
ChaCha.java: Update to include YYYY-MM-DD format dates
EventTask.java: Update date field to be of LocalDate type
DeadlineTask.java: Update date field to be of LocalDate type
Task.java: Update date field to be of LocalDate type
ListOfTask.java: Update the way chacha.txt is updated with new information

SherniceSng committed:
Merge branch 'branch-Level-8'

SherniceSng committed:
Add and update files to allow chatbot to follow OOP better

SherniceSng committed:
Fix the bug which happens when overwritting chacha.txt

SherniceSng committed:
Clean up the code, remove unused files and codes

SherniceSng committed:
Merge branch 'branch-A-MoreOOP'

SherniceSng committed:
Organise classes into package chacha

SherniceSng committed:
Merge branch 'branch-A-Packages'

SherniceSng committed:
Update to follow Coding Standard

SherniceSng committed:
Update the formatting of code

SherniceSng committed:
Update to include JavaDoc comments

SherniceSng committed:
Update code to allow users to search a keyword

SherniceSng committed:
Merge branch 'branch-Level-9'

SherniceSng committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/chacha/Ui.java

SherniceSng committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/chacha/Parser.java
#	src/main/java/chacha/Ui.java

SherniceSng committed:
Merge branch 'add-gradle-support'

SherniceSng committed:
Add JavaDoc comments in Parser.java and TaskList.java

SherniceSng committed:
Add ParserTest.java and TaskList.java to test Parser class and TaskList class

SherniceSng committed:
Update build.gradle

yu7ong commented on src/main/java/chacha/Parser.java:
```diff
@@ -0,0 +1,200 @@
+ */
+public class Parser {
+    private Storage storage;
+    private TaskList tasks;
```
The use of plural to represent a collection of objects is good

yu7ong commented on src/main/java/chacha/Parser.java:
```diff
@@ -0,0 +1,200 @@
+            String[] arr = {"Hmmm... I don't seem to understand this. Please input another command! "};
+            return this.ui.printStrings(arr);
+        }
+    }
```
I like how the method names follow the standard camelCase conventions.

yu7ong commented on src/main/java/chacha/Parser.java:
```diff
@@ -0,0 +1,200 @@
+     *
+     * @return String representation.
+     */
+    public String doAddToDoCommand(String userInput) {
```
Perhaps it will be clearer to shorten the name to "addTodo" instead?

yu7ong commented on src/main/java/chacha/Storage.java:
```diff
@@ -0,0 +1,107 @@
+
+            while ((line = this.readerFile.readLine()) != null) {
+                String[] arr = line.split(" \\| ");
+                boolean isDone = (arr[1].equals("1"));
```
The naming of boolean is good

yu7ong reviewed (commented):
Looks good!

SherniceSng committed:
Separate commands into different classes

SherniceSng committed:
Add GUI to the chatbot

SherniceSng committed:
Merge branch 'Level-10'

natlv commented on src/main/java/chacha/DialogBox.java:
```diff
@@ -0,0 +1,62 @@
+
+        dialog.setText(text);
+        displayPicture.setImage(img);
+        Circle clip = new Circle(displayPicture.getFitWidth() / 2, displayPicture.getFitHeight() / 2, displayPicture.getFitWidth() / 2);
```
Perhaps 'image' or 'picture' would be a better variable name than 'clip' here? 'clip' could imply a video

natlv reviewed (commented):
LGTM, I think your program structure is really good. Just some nits to fix

SherniceSng committed:
Update the styles of files

SherniceSng committed:
Merge branch 'branch-A-CheckStyle'

SherniceSng committed:
Add assertions to code

SherniceSng committed:
Modify to remove unused parts of code

SherniceSng committed:
Update to improve code quality

Many of the exception to users have messages in common.

The common messages across messages causes code duplication in
TaskList.java.

Extracting some of the common messages allows less repetition of the
same messages within the same class.

Let's,
* add WrongTimeFormatException.java to encapsulate common messages
relating to time fields
* update ChaChaException.java to not take in a message to reduce code
duplication
* add common messages as constants to be returned after
ChaChaException is caught

SherniceSng committed:
Merge pull request #1 from SherniceSng/branch-A-Assertions

Add assertions to code

SherniceSng committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/chacha/ChaCha.java
#	src/main/java/chacha/Ui.java
#	src/main/java/chacha/command/Parser.java
#	src/main/java/chacha/data/chacha.txt

SherniceSng committed:
Help command: give guidance to users

There is no instructions for users to help them with the application.

Encapsulating all the messages into a class allows us to call upon
different help messages in response to different user inputs.

Let's add HelpCommand.java that encapsulates the help messages for
the user.

SherniceSng committed:
Merge pull request #2 from SherniceSng/branch-C-Help

Help command: give guidance to users

SherniceSng committed:
Remove binary files

SherniceSng committed:
Update EventTask with valid time format

SherniceSng committed:
Add JavaDoc comments, improve code quality

SherniceSng committed:
Update README.md

SherniceSng committed:
Add unit testing files

SherniceSng committed:
Add padding and colour in GUI

SherniceSng committed:
Update user guide

SherniceSng committed:
Fix Ui.png in user guide

SherniceSng committed:
Make small changes to user guide

SherniceSng committed:
Make some changes to user guide

SherniceSng committed:
Update Javadoc comments, make some changes

SherniceSng committed:
Update comments

SherniceSng committed:
Update HelpCommand.java
